File: InitialProofOfConcept/InitialProofOfConcept/AKTable.m
//
//  AKTable.m
//  AudioKit
//
//  Created by Aurelius Prochazka on 9/3/15.
//  Copyright (c) 2015 Aurelius Prochazka. All rights reserved.
//

// implement variable size and different standard waveforms and such

#import "AKTable.h"

@implementation AKTable {
    sp_ftbl *ft;
}

- (instancetype)init
{
    self = [super init];
    if (self) {
        [self create];
    }
    return self;
}
- (void)create
{
    AKManager *manager = [AKManager sharedManager];
    sp_ftbl_create(manager.data, &ft, 4096);
    sp_gen_sine(manager.data, ft);
}

- (void)destroy
{
    sp_ftbl_destroy(&ft);
}

- (sp_ftbl *)table
{
    return ft;
}


@end

File: SwiftOSXProofOfConcept/SwiftOSXProofOfConcept/AKTable.swift
//
//  AKTable.swift
//  AudioKit
//
//  Created by Aurelius Prochazka on 9/5/15.
//  Copyright (c) 2015 Aurelius Prochazka. All rights reserved.
//

import Foundation

class AKTable : AKParameter {
    var ftbl: UnsafeMutablePointer<sp_ftbl> = nil  //not just nil
    
    override init() {
        super.init()
        create()
    }
    
    func create() {
        sp_ftbl_create(AKManager.sharedManager.data, &ftbl, 4096)
        sp_gen_sine(AKManager.sharedManager.data, ftbl);
    }
    
    func destroy() {
        sp_ftbl_destroy(&ftbl)
    }

}